iocage/cli/list.py

```diff
@@ -42,52 +43,78 @@
 @click.option("--remote", "-R", is_flag=True, help="Show remote's available "
                                                    "RELEASEs.")
 @click.option("--plugins", "-P", is_flag=True, help="Show available plugins.")
-@click.option("--http", default=False,
-              help="Have --remote use HTTP instead.", is_flag=True)
 @click.option("--sort", "-s", "_sort", default="name", nargs=1,
               help="Sorts the list by the given type")
 @click.option("--quick", "-q", is_flag=True, default=False,
               help="Lists all jails with less processing and fields.")
-def cli(dataset_type, header, _long, remote, http, plugins, _sort, quick):
-    """This passes the arg and calls the jail_datasets function."""
-    freebsd_version = ioc_common.checkoutput(["freebsd-version"])
-    iocage = ioc.IOCage(skip_jails=True)
+@click.option("--log-level", "-d", default="info")
+@click.option("--output", "-o", default=None)
+def cli(dataset_type, header, _long, remote, plugins,
+        _sort, quick, log_level, output):
+
+    logger = Logger.Logger(print_level=log_level)
+    host = Host.Host(logger=logger)
+    jails = Jails.Jails(logger=logger)
 
     if dataset_type is None:
         dataset_type = "all"
 
     if remote and not plugins:
-        if "HBSD" in freebsd_version:
-            hardened = True
-        else:
-            hardened = False
 
-        ioc_fetch.IOCFetch("", http=http, hardened=hardened).fetch_release(
-            _list=True)
+        available_releases = host.distribution.releases
+        for available_release in available_releases:
+            print(available_release.name)
```

Comment:
The logger needs the ability to provide custom callback methods and to disable the built-in printing. For library usage, I'd even disable it by default. A custom log method could then handle logging to a remote server, local files or what else one can do with the logs.